subprojects/core/src/main/java/org/gradle/process/internal/JavaExecHandleBuilder.java

```diff
@@ -207,7 +207,10 @@ public JavaExecSpec args(Iterable<?> args) {
     }
 
     public JavaExecHandleBuilder setClasspath(FileCollection classpath) {
-        doGetClasspath().setFrom(classpath);
+        // Resetting the field as otherwise there can be reference loop issue - See gradle/gradle#8748
+        ConfigurableFileCollection newClasspath = fileCollectionFactory.configurableFiles("classpath");
```

Comment:
Probably should add a test that adding the classpath to itself doesn't blow up, so we remember about this problem in the future.

Reply:
Could you also fix `classpath()` so that it calls `from()` instead of `setFrom()`?

Reply:
Will add the test, makes sense.

As for the other issue, was fixed in a previous PR already merged - see #8752